src/main/scala/mesosphere/marathon/api/v2/json/V2AppDefinition.scala

```diff
@@ -2,17 +2,23 @@ package mesosphere.marathon.api.v2.json
 
 import java.lang.{ Double => JDouble, Integer => JInt }
 
+import com.wix.accord._
 import mesosphere.marathon.Protos.Constraint
+import mesosphere.marathon.Protos.HealthCheckDefinition.Protocol
+import mesosphere.marathon.api.v2.Validation._
 import mesosphere.marathon.api.validation.FieldConstraints._
 import mesosphere.marathon.api.validation.{ PortIndices, ValidV2AppDefinition }
 import mesosphere.marathon.health.HealthCheck
 import mesosphere.marathon.state.AppDefinition.VersionInfo.FullVersionInfo
 import mesosphere.marathon.state._
+import mesosphere.marathon.state.PathId._
 import org.apache.mesos.{ Protos => mesos }
 
 import scala.collection.immutable.Seq
 import scala.concurrent.duration._
 
+import com.wix.accord.dsl._
+
 @PortIndices
```

Comment:
@aquamatthias Removing annotation validation, I realised, that there is also a HealthCheck validation. I implemented that one. Still there is a comment which caught my attention:
```scala
/**
    * FIXME: This is not called when an V2AppDefinition is called because the recursion apparently doesn't work
    * with Scala Seq.
    *
    * We cannot re-enable it because it would break backwards compatibility in weird ways. If users hard already
    * one invalid app definition, each deployment would cause a complete revalidation of the root group
    * including the invalid one. Until the user changed all invalid apps, the user would get weird validation
    * errors for every deployment potentially unrelated to the deployed apps.
    */
```
That comment is regarding HealthCheck validation. Could you get back to me on that?